### PR TITLE
Add  dataset

### PR DESCRIPTION
**Pull request by fdsa654hg**
add english to chinese dataset

### CODE REVIEW

**Review by ProFatXuanAll — CHANGES_REQUESTED** [review]
Need to add a dataset class, see `s2s.dset` for examples.

File: s2s/dset/_eng2chi.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        super().__init__()
+        df = pd.read_csv(os.path.join(DATA_PATH, 'eng2chi.csv'))
+        self.src.extend(
+            df['src'].apply(str).apply(self.__class__.preprocess).to_list()
+        )
+        self.tgt.extend(
+            df['tgt'].apply(str).apply(self.__class__.preprocess).to_list()
+        )
+
+    @staticmethod
```
Comment (ProFatXuanAll) [inline s2s/dset/_eng2chi.py:55]: For translation, one should use evaluation metrics other than exact match (i.e., generate words must "be the same word" and "in the same order" of target sequence , which is used by `accuracy_score`).
Why is exact match bad?
Consider the example translation pair `I like apple` and `我喜歡蘋果`, `蘋果我喜歡` should also be an acceptable answer.
Thus, one should use some evaluation metrics with **fozzy match**, i.e., it's okay to not be *that* accurate but at the same time have same meaning (swapping order, synonym, etc.).
Nowadays people mostly use **BLEU score** as evaluation metric on translation task.
Go find some python package which calculate BLEU score for you.

File: s2s/dset/_eng2chi.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+from typing import Sequence
+
+import pandas as pd
+from sklearn.metrics import accuracy_score
+
+from s2s.dset._base import BaseDset
+from s2s.path import DATA_PATH
+
+class Eng2ChiDset(BaseDset):
```
Comment (ProFatXuanAll) [inline s2s/dset/_eng2chi.py:12]: Add docstring for `Eng2ChiDset` class, including reference of the source of the dataset.

**Review by ProFatXuanAll — CHANGES_REQUESTED** [review]

File: Pipfile [hunk 3]
```diff
@@ -14,6 +14,7 @@ numpy = "1.19.2"
 tqdm = "4.49.0"
 sklearn = "0.0"
 tensorboard = "2.3.0"
+nltk = "*"
```
Comment (ProFatXuanAll) [inline Pipfile]: Specify dependency version

File: run_eval_model.py [hunk 4]
```diff
@@ -124,6 +124,10 @@ def main():
         batch_tgt=dset.all_tgt(),
         batch_pred=all_pred,
     ))
+    print(DSET_OPTS[args.dset_name].bleu_score(
```
Comment (ProFatXuanAll) [inline run_eval_model.py]: This method only work on translation dataset but not arithmetic dataset.
Overload `batch_eval` can do the trick.

File: s2s/dset/_eng2chi.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+from typing import Sequence
+
+import pandas as pd
+from sklearn.metrics import accuracy_score
+from nltk.translate.bleu_score import sentence_bleu
+
+from s2s.dset._base import BaseDset
+from s2s.path import DATA_PATH
+
+try:
```
Comment (ProFatXuanAll) [inline s2s/dset/_eng2chi.py]: User need to install dependency before running the code.
So you don't need to check whether a dependecies is missing or not.
